8.x/theme-development/helpers.md

```diff
@@ -1202,418 +1650,848 @@ This function returns all the customer invoices that are logged in.
+This function is practical for creating suggestions.
 
+----
 
+#### get_related_products_with_directory_level
 
-### `get_product_similar_products`
+```php
+function get_related_products_with_directory_level(Product $product, int $count = 5, int $level = 1) {
+        $directory = $product->directory;
+        if ($level > 1) {
+            while ($level != 1) {
+                $directory = $directory->parentDirectory;
+                $level--;
+            }
+        }
+        return $directory->leafProducts()->mainModels()->visible()->except($product->id)
+            ->orderBy("priority", "ASC")->take($count)->get();
+    }
+```
 
-The `get_product_similar_products` function declares a similarity key for a product. For example, a car with the BMW brand is considered. This function returns products with the exact specifications as this car but with a different brand.
+This function returns related products within the same category. To maintain similarity, we recommend searching for products in one level higher within the category hierarchy (but not beyond).
 __________________________
 
+#### get_product_attributes
 
+```php
+function get_product_attributes(Product $product = null) {
+        if ($product != null) {
+            $attributes = PAttr::getProductAttributes($product);
+            return $attributes['attributes'];
+        } else {
+            return null;
+        }
+    }
+```
+This function returns a product's related attributes such as its brand, maker, country, material, and all the product's keys and values.
 
+----
 
-### `get_related_products_with_directory_level`
+#### get_article_related_products
+```php
+ function get_article_related_products(Article $article, int $count = 3) {
+        $tags = $article->tags()->get()->pluck('id');
+        return Product::mainModels()->visible()->whereHas('tags', function ($query) use ($tags) {
+            $query->whereIn('id', $tags);
+        })->where("is_active", true)->mainModels()->visible()->latest()->take($count)->get();
+    }
+```
 
-Returns related products having the same parent in which they are in specific higher levels passed by `$level`.
-__________________________
+This function considers an article and returns the related products. Articles and products connection method take parts with an object name `tags`.
+**Note:** This is completely different from Seo tags.
 
+----
 
+#### get_article_related_articles
 
-### `get_product_most_privileged_key_attributes`
+```php
+ function get_article_related_articles(Article $article, int $count = 4) {
+        return $article->directory->articles()->latest()->except($article->id)->take($count)->get();
+    }
+```
 
-This function takes a list of all the keys and then returns the validity of that key. This function will be deprecated in the next version.
-__________________________
+This function return related article to the given article based on their tags.
```

Comment:
This function return related article to the given article -> This function returns the articles related to a given article